test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -119,6 +121,7 @@ public void AutoFitRegressionTest()
                     new ColumnInformation() { LabelColumnName = DatasetUtil.MlNetGeneratedRegressionLabel });
 
             Assert.True(result.RunDetails.Max(i => i.ValidationMetrics.RSquared > 0.9));
+            result.Dispose();
```

Comment:
A user will need to return their model from a function like this, and the `result.Dispose()` will dispose of it (and all the other models) before it can be returned. 

Can you modify this unit test run within a function which returns the one best model to the parent function, and cleans up the rest of the models trained?

Reply:
I believe the `bestRun `model is stored as a copy of the model of the best run in `runDetails`. So would it be enough to only Dispose of the models in `runDetails `and not dispose of `bestRun`'s model?
  
Here are the hash codes of the models of the two models, one is in `bestRun `and one is the only model in `runDetails`. As the hash codes are different, they should be distinct (but the same) models. `Object.ReferenceEquals` also confirms this.
![diff](https://user-images.githubusercontent.com/5262061/79029720-7ccfa200-7b4a-11ea-8d48-df165cbc27d8.png)